include/pybind11/pytypes.h

```diff
@@ -224,6 +224,18 @@ inline handle get_function(handle value) {
     return value;
 }
 
+// Helper aliases/functions to support implicit casting of values given to python accessors/methods.
+// When given a pyobject, this simply returns the pyobject as-is; for other C++ type, the value goes
+// through pybind11::cast(obj) to convert it to an `object`.
+template <typename T, enable_if_t<is_pyobject<T>::value, int> = 0>
+T object_or_cast(T o) { return o; }
+// The following casting version is implemented in cast.h:
+template <typename T, enable_if_t<!is_pyobject<T>::value && !std::is_same<T, PyObject*>::value, int> = 0>
+object object_or_cast(const T &o);
+// Match a PyObject*, which we want to convert directly to handle via its converting constructor
+template <typename T, enable_if_t<std::is_same<T, PyObject>::value, int> = 0> handle object_or_cast(T *ptr) { return ptr; }
```

Comment:
It wasn't clear to me why the return value is ``handle`` here. Could the ``reinterpret_borrow`` be avoided below if we return an object here? Do we even need the ``PyObject*`` version?